src/kOS.Safe/Exceptions/KOSKeyNotFoundException.cs

```diff
@@ -0,0 +1,12 @@
+﻿namespace kOS.Safe.Exceptions
+{
+    public class KOSKeyNotFoundException : KOSException
+    {
+        private const string MSG = "The given key {0} was not present, use CONTAINS for checking safety. This is a case {1} collection";
+
+        public KOSKeyNotFoundException(string key, bool caseSensitive) : base(string.Format(MSG, key, caseSensitive ? "insensitive" : "sensitive"))
```

Comment:
I'm slightly wary that stating "This is a case [in]sensitive collection" during the error message may trick users into thinking that kOS is trying to tell them this is the reason for their error.  It's printing it unconditionally regardless of whether or not that had anything to do with it.  It makes sense to print it when they looked for "cat" and the key was really "Cat", but not when they looked for "cat" and the key was really "dog".

At the very least, can it only be printed when it's sensitive?  It cannot possibly be the cause of error when it's insensitive.  (And the reverse for the duplicate key exception above, where case can only cause a duplication error when it's insensitive.)

Or maybe just not make it a separate sentence but just make it part of the description of the thing.  That will make it look like less of a claim that it's the cause of the error, while still hinting people that it might be.

i.e. "The given key "{0}" was not present in the case-{sensitive/insensitive} collection.  Use CONTAINS for checking safety".